src/clientiface.cpp

```diff
@@ -335,6 +325,60 @@ void RemoteClient::GetNextBlocks (
 				continue;
 			}
 
+			/*
+				If block is not close, don't send it unless it is near
+				ground level.
+
+				Block is near ground level if night-time mesh
+				differs from day-time mesh.
+			*/
+			if (block && d >= d_opt && !block->getDayNightDiff()) {
+				// getDayNightDiff() does not take adjacent blocks into account.
+				// If necessary, remember this block, so that it may be sent anyway if possible.
+				if (extra_blockpos.count(p))
+					continue;
+				int extra = -1;
+				do {		// Can break ot of this
```

Comment:
IDK.
IMO there is more structure to this, and it is therefore easier to comprehend.